src/events/deposits.controller.ts

```diff
@@ -62,15 +63,9 @@ export class DepositsController {
       }),
     )
     data: UpsertDepositsDto,
-  ): Promise<SerializedDeposit[]> {
-    const deposits = await this.depositsUpsert.upsertBulk(data.operations);
-
-    return deposits.map((d) => ({
-      id: d.id,
-      transaction_hash: d.transaction_hash,
-      block_sequence: d.block_sequence,
-      block_hash: d.block_hash,
-      object: 'deposit',
-    }));
+    @Res() res: Response,
+  ): Promise<void> {
+    await this.depositsUpsert.bulkUpsert(data.operations);
```

Comment:
why await?

Reply:
We need an `await` here to propagate exceptions and errors properly to the NestJS [execution context](https://github.com/nestjs/nest/blob/master/packages/core/router/router-execution-context.ts#L139-L149). If an error happened without this `await`, the cluster would [exit](https://github.com/iron-fish/ironfish-api/blob/master/src/main.ts#L116-L122) and re-fork. With the `await`, NestJS properly sees exceptions and throws a 500.

Additionally, having these `await`s helps with timing and monitoring for the statistics we send over to Datadog.